### PR TITLE
Add readme section for good practices

### PR DESCRIPTION
**Pull request by bishal-pdMSFT**
There are certain solutions which allow to use cache for some interesting use cases (e.g. #171). Adding a section to highlight such use cases and potential good way to solve them.

### CODE REVIEW

**Comment by bishal-pdMSFT** [comment]
#840

File: README.md [hunk 1]
```diff
@@ -219,6 +219,23 @@ jobs:
...
+        uses: actions/cache@v3
+        with:
+          path: prime-numbers
+          key: primes-${{ runner.os }}-${{ github.run_id }} # Can use time based key as well
+          restore-keys: |
+            primes-${{ runner.os }}
+  ```          
+  Please note that this will create a new cache on every run and hence will consume the cache [quota](#cache-limits).
+  
+- **Use cache across feature branches** - Reusing cache across feature branches is not allowed today to provide cache [isolation](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache). However if both feature branches are from same base branch, a good way to achieve this is to ensure that the base branch has a cache. This cache will then be consumable by both feature branches.
```
Comment (kotewar) [inline README.md:237]: > same base branch

Shall we specifically mention that the base branch here is only the default branch? I think this won't work if the base branch is a feature branch.

Comment (bishal-pdMSFT) [reply README.md:237]: @kotewar this is what docs say

>Also, if the branch feature-b has the base branch feature-a, a workflow triggered on feature-b would have access to caches created in the default branch (main), feature-a, and feature-b

**Review by kotewar — APPROVED** [review]
LGTM

**Pull request merged**